src/YieldDistributor.sol

```diff
@@ -73,6 +73,16 @@ contract YieldDistributor is OwnableUpgradeable {
     address[] public queuedProjectsForRemoval;
     // @notice The voting power allocated to projects by voters in the current cycle
     uint256[] public projectDistributions;
+    /* IIUC, `projectDistributions` and `projects` are intended to be kept in sync.
+    - This is a somewhat fragile pattern. At the very least, it should be commented and made explicit.
+    - It would be better to have a single array with a struct holding the address and uint256. That is
+      not gas-optimal, but given that this contract lives on Gnosis I think the balance between code
+      quality and gas optimization should tilt heavily towards the former.
+    - Even better IMO would be a `mapping(address => uint256)` to track distributions per project.
+      That would allow a voting interface that passes votes by address rather than array index, which
+      is pretty hard to debug after the fact (e.g. through a chain explorer). The current setup is more
+      vulnerable to either of an evil frontend or MEV. Neither of those is a proximate issue, but still.
```

Comment:
It occurs to me that more probable than an evil frontend is a bug in the real frontend that causes the displayed project order to mismatch with the contract project order. Protecting against that seems worthwhile.